File: flutter/flutter-sdk-bridge/ping_journey/android/src/main/kotlin/com/pingidentity/flutter/journey/PingJourneyPlugin.kt
/*
 * Copyright (c) 2026 Ping Identity Corporation. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

package com.pingidentity.flutter.journey

import io.flutter.embedding.engine.plugins.FlutterPlugin

/** Registers the generated [PingJourneyHostApi] Pigeon channel. */
class PingJourneyPlugin : FlutterPlugin {
    private var api: JourneyHostApiImpl? = null

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        api = JourneyHostApiImpl()
        PingJourneyHostApi.setUp(binding.binaryMessenger, api)
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        PingJourneyHostApi.setUp(binding.binaryMessenger, null)
        api?.shutdown()
        api = null
    }
}

File: flutter/flutter-sdk-bridge/ping_journey/ios/ping_journey/Sources/ping_journey/PingJourneyPlugin.swift
/*
 * Copyright (c) 2026 Ping Identity Corporation. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

import Flutter
import UIKit

/// Registers the generated `PingJourneyHostApi` Pigeon channel.
public class PingJourneyPlugin: NSObject, FlutterPlugin {
    private var api: JourneyHostApiImpl?

    public static func register(with registrar: FlutterPluginRegistrar) {
        let instance = PingJourneyPlugin()
        let api = JourneyHostApiImpl()
        instance.api = api
        PingJourneyHostApiSetup.setUp(binaryMessenger: registrar.messenger(), api: api)
        registrar.publish(instance)
    }

    public func detachFromEngine(for registrar: FlutterPluginRegistrar) {
        PingJourneyHostApiSetup.setUp(binaryMessenger: registrar.messenger(), api: nil)
        api = nil
    }
}
